Bug 1896964 - Cleanup firefox-android repo references

gbrownmozilla opened the pull request:
Remove all references to the firefox-android and staging-firefox-android repos.

gbrownmozilla committed:
Bug 1896964 - Cleanup firefox-android repo references

gbrownmozilla committed:
Bug 1896964 - Cleanup firefox-android repo references

gbrownmozilla committed:
more cleanup, mostly tests and examples

gbrownmozilla committed:
more cleanup, mostly tests and examples

gbrownmozilla commented on githubscript/examples/config.example.json:
```diff
@@ -1,15 +1,15 @@
           "github_token": "SOME_TOKEN",
           "github_owner": "mozilla-mobile",
-          "github_repo_name": "firefox-android",
+          "github_repo_name": "reference-browser",
```
reference-browser does not actually use githubscript. I've replaced firefox-android with reference-browser in examples and tests just to provide the name of an active github repo; open to suggestions for a better approach!

jcristau replied:
Should we remove the mobile bits entirely and use an example from xpi instead?

gbrownmozilla replied:
Yes!

jcristau commented on beetmoverscript/docker.d/worker.yml:
```diff
@@ -11,8 +11,6 @@ taskcluster_scope_prefixes:
         - 'project:releng:beetmover:'
       'COT_PRODUCT == "thunderbird"':
         - 'project:comm:thunderbird:releng:beetmover:'
-      'COT_PRODUCT == "mobile"':
-        - 'project:mobile:firefox-android:releng:beetmover:'
```
We should also remove all the other mobile cot-product bits in this file and init_worker.sh.

gbrownmozilla replied:
Thanks. I think I have those cleaned up now.

jcristau commented on signingscript/docker.d/passwords.yml:
```diff
@@ -78,42 +78,6 @@ in:
-             ["autograph_gpg"]
-          ]
-          - ["https://autograph-external.prod.autograph.services.mozaws.net",
-             {"$eval": "AUTOGRAPH_FOCUS_USERNAME"},
-             {"$eval": "AUTOGRAPH_FOCUS_PASSWORD"},
```
init_worker.sh should be updated to not require variables we're no longer using.

gbrownmozilla replied:
Done. Also noticed/remembered pushapkscript.

jcristau commented on treescript/src/treescript/gecko/versionmanip.py:
```diff
@@ -31,9 +31,7 @@
-    "fenix": MobileVersion,
-    "firefox-android": MobileVersion,
-    "focus-android": MobileVersion,
+    "placeholder-repo": MobileVersion,  # not currently used
```
Any reason to not just remove the unused code?

gbrownmozilla replied:
Sorry, I'm not sure I understand what you had in mind. I removed that `placeholder-repo` line - it didn't add value. Were you thinking of removing `_VERSION_CLASS_PER_END_OF_SOURCE_REPO`? I've left it thinking that it's generic and might be useful again.
Of course `treescript/github` is also entirely unused now, but I think that's useful in the foreseeable future. Removing that, or cleaning up the github tests further, seems to get a little complicated; I wouldn't mind leaving that for a follow-up.

jcristau reviewed (changes requested):
Thanks for cleaning this up :)

gbrownmozilla committed:
cleanup more mobile entries in docker.d directories

gbrownmozilla committed:
cleanup more mobile entries in docker.d directories

gbrownmozilla committed:
fix test_init.py

gbrownmozilla committed:
fix test_init.py

gbrownmozilla committed:
use xpi for githubscript examples

gbrownmozilla committed:
use xpi for githubscript examples

gbrownmozilla committed:
fix-up config.example.json

gbrownmozilla committed:
fix-up config.example.json

gbrownmozilla committed:
tweak treescript versionmanip

gbrownmozilla committed:
tweak treescript versionmanip

pre-commit-ci[bot] committed:
style: pre-commit.ci auto fixes [...]

pre-commit-ci[bot] committed:
style: pre-commit.ci auto fixes [...]

bhearsum committed:
Bump dependencies (#1002)

jcristau commented on tests/test_init.py:
```diff
@@ -256,6 +256,7 @@ def generate_params():
     xfail = {
         "pushapk-firefox-dev",
         "pushapk-firefox-fake-prod",
+        "pushapk-firefox-prod",
```
Curious what the problem with these is :)

gbrownmozilla replied:
Me too! It wasn't just one problem, of course. Fixed now.

gbrownmozilla replied:
Actually, it seemed fixed locally, but wasn't reliable in all environments. I found *another* problem - pushapk's init_worker relies on keytool - so I restored the xfails and added a comment.

jcristau reviewed (approved):
Thanks.

I think we'll want to remove the beetmover mobile pools and the githubscript mobile prod pool from https://github.com/mozilla-services/cloudops-infra/blob/master/projects/relengworker/Jenkinsfile and https://github.com/mozilla-services/cloudops-infra/blob/master/projects/relengworker/Jenkinsfile.dev before deploying this, to avoid failed deploys.

gbrownmozilla committed:
fix and enable the pushapkscript init tests

gbrownmozilla committed:
fix and enable the pushapkscript init tests

gbrownmozilla committed:
Merge branch 'no-firefox-android' of https://github.com/gbrownmozilla/scriptworker-scripts into no-firefox-android

hneiva committed:
feat(bitrisescript): Resume job if possible (#978)

jcristau committed:
Bump dependencies (#1004)

New mozilla-version and scriptworker releases add support for ESR128.

jcristau committed:
scriptworker_client: also run tests on python 3.11

Assumes dependencies are the same or close enough to those on 3.9, at
least until we move everything from 3.9 to 3.11.

jcristau committed:
scriptworker_client: fix TypeError on python 3.11

Calling `random.sample()` on a dict_keys object no longer works

```
FAILED tests/test_aio.py::test_LockfileFuture[True] - TypeError: Population must be a sequence.  For dicts or sets, use sorted(d).
FAILED tests/test_aio.py::test_LockfileFuture[False] - TypeError: Population must be a sequence.  For dicts or sets, use sorted(d).
```

https://docs.python.org/3/whatsnew/3.11.html#porting-to-python-3-11

jcristau committed:
scriptworker_client: fix TypeError in test_pipe_to_log on python 3.11

```
FAILED tests/test_utils.py::test_pipe_to_log - TypeError: Passing coroutines is forbidden, use tasks explicitly.
```

Per https://docs.python.org/3.11/library/asyncio-task.html#asyncio.wait:
"Changed in version 3.11: Passing coroutine objects to wait() directly is forbidden."

jcristau committed:
ci: add python 3.11 base images (#1009)

jcristau committed:
balrogscript: delete old tc.yml (#1007)

pre-commit-ci[bot] committed:
chore: pre-commit autoupdate (#1003)

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.4.4 → v0.4.5](https://github.com/astral-sh/ruff-pre-commit/compare/v0.4.4...v0.4.5)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

ahal committed:
fix(treescript): use separate Github apps for prod vs dev

Initially I didn't bother splitting these out because I figured they
both require the exact same permissions on the repo anyway. So if the
dev app secret were compromised, you'd be just as hosed as if the prod
app secret were compromised.

However, it didn't occur to me that we might only install the `dev` app
on staging repos and that we don't want a bug on treescript-dev to
accidentally expose the `prod` app's private key.

To this end, I created a `releng-treescript-dev` app. We should probably
avoid installing this app on production repos, and instead only add it
to `staging-*` repos. To help with this, I restricted the app to the
`mozilla-releng` org.

jcristau committed:
maintenance: add support for pinning dependencies with python 3.11 (#1011)

jcristau committed:
configloader: test with python 3.11 (#1014)

This assumes dependencies are the same or close enough as on python 3.9.

jcristau committed:
addonscript: run on python 3.11 (#1006)

jcristau committed:
ci: make pr-complete depend on lint tasks (#1016)

jcristau committed:
balrogscript: run on python 3.11 (#1008)

jcristau committed:
beetmoverscript: run on python 3.11 (#1012)

barthalion committed:
pushflatpakscript: Run on python 3.11, update flat-manager-client to 0.4.2 (#1018)

* pushflatpakscript: Run on python 3.11, update flat-manager-client to 0.4.2

* pushflatpakscript: more changes to switch to python 3.11

---------

Co-authored-by: Julien Cristau <jcristau@mozilla.com>

bhearsum committed:
Bump dependencies (#1020)

No major bumps today.

bhearsum committed:
treescript: add support for fetching l10n revisions in a github repo (#1015)

* feat(treescript): Add support for fetching head revision with github client

Mostly a refactor of existing code.

* feat(treescript): Allow git and hg configuration simultaneously

* feat(treescript): Enable git for Firefox treescript

* refactor(treescript): Only bump l10n in merge day action when an l10n_bump action is also present

This moves the decision about bumping l10n up to a higher level, which simplifies adding support for bumpning via github in the next commit.

* feat(treescript): Add support for fetching l10n revisions from github.

Most of this is copy/paste to avoid touching any of the existing l10n bumping code.

The most notable change is in build_revision_dict_github, which is vastly simplified compared to its hg compared part. In part this is because we only have one string repo to worry about, but also because @eemeli declared that we don't need to support pinning revisions anymore - so I didn't implement that here. After we cut over to the Github repo and know we aren't going back, we can rip out the hg support.

Most of the rest of this is just wiring in calls to l10n_bump_github when appropriate.

* force treestatus check off for testing

* style: pre-commit.ci auto fixes [...]

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

gbrownmozilla committed:
Merge branch 'master' into no-firefox-android

gbrownmozilla committed:
Merge branch 'master' into no-firefox-android

gbrownmozilla committed:
Merge branch 'no-firefox-android' of https://github.com/gbrownmozilla/scriptworker-scripts into no-firefox-android

gbrownmozilla committed:
"fix" test_init for pushapk, again

hneiva commented on githubscript/examples/task.example.json:
```diff
@@ -1,99 +1,73 @@
     "checks"
   ],
-  "priority": "highest",
+  "priority": "lowest",
```
heheh this is a good idea :)

hneiva reviewed (approved):
LGTM
